src/app/[locale]/selected-filters.tsx

```diff
@@ -0,0 +1,114 @@
+import {SearchResultFilter} from '@/lib/dataset-fetcher';
+import {Dispatch} from 'react';
+import {useTranslations} from 'next-intl';
+import {XMarkIcon} from '@heroicons/react/24/solid';
+
+interface TagProps {
+  name?: string;
+  remove: () => void;
+}
+
+function Tag({name, remove}: TagProps) {
```

Comment:
What is your preferred strategy regarding components-in-components? This file could get rather large. At which point should we move inner components to their own files (e.g. for readability/maintainability/testability)?

Reply:
This is a good question. In the past, I have put a lot of thought into how to structure reusable components. But that is more for component systems and less for single-use components.

Single-use components like this one I like to keep as close to the parent component as possible. So I prefer keeping it in the same file and only moving it if the child component gets too big. But I know that 'big' is subjective.

In this case, Tag is only markup and has no logic, so I consider this 'small' and only used by the selected filters, therefor I keep it in the selected-filters file.

Reply:
In this case, I will create a more generic Tag component in the component folder.

Reply:
I decided to rename Tag to Badge because it is based on a tailwind UI badge. And refactored the badge to a more dynamic component. I placed the generic component in the components directory.

Reply:
Thanks! Do you already know we need the Badge as a generic component? I think it's helpful. But if we don't know yet, we could keep it in the `app` directory as a specific component and promote it to a generic one if required.

Reply:
Discussed offline: The design already shows two places with a Badge. So we can keep this generic.